app/javascript/admin/controllers/consumer_app_controller.js

```diff
@@ -0,0 +1,17 @@
+import { Controller } from 'stimulus';
+
+export default class ConsumerAppController extends Controller {
+  static targets = ['platform', 'teamId'];
+
+  connect() {
+    this.checkPlatform();
+  }
+
+  checkPlatform() {
+    if (this.platformTarget.value === 'ios') {
+      this.teamIdTarget.classList.remove('hidden');
+    } else {
+      this.teamIdTarget.classList.add('hidden');
+    }
+  }
```

Comment:
This very simple Stimulus controller provides the show/hide functionality for the `ConsumerApp` form. This way when the creator is creating/editing a `ConsumerApp` they will only see the `Team ID` input if they're dealing with iOS and the opposite for Android.

This is tested in `cypress/integration/adminFlows/consumerApps/createConsumerApp.spec.js`